src/node_api_helpers.h

```diff
@@ -205,30 +190,25 @@ namespace Napi {
       HandleScope scope;
       napi_env env;
       napi_get_current_env(&env);
-      napi_value h;
-      napi_get_reference_value(env, handle, &h);
-      napi_set_element(env, h, kCallbackIndex, fn);
+      napi_delete_reference(env, handle);
```

Comment:
Only because of the `CHECK_ARG` in `napi_delete_reference` (is that actually necessary?). Calling `delete` on `nullptr` is perfectly fine.